pkg/controller/deployment/deployment_controller.go

```diff
@@ -1176,11 +1294,15 @@ func (dc *DeploymentController) scaleDownOldReplicaSetsForRecreate(oldRSs []*ext
 
 // scaleUpNewReplicaSetForRecreate scales up new replica set when deployment strategy is "Recreate"
 func (dc *DeploymentController) scaleUpNewReplicaSetForRecreate(newRS *extensions.ReplicaSet, deployment *extensions.Deployment) (bool, error) {
+	glog.V(4).Infof("Scaling up new replica set %q from %d to %d", newRS.Name, newRS.Spec.Replicas, deployment.Spec.Replicas)
 	scaled, _, err := dc.scaleReplicaSetAndRecordEvent(newRS, deployment.Spec.Replicas, deployment)
 	return scaled, err
 }
 
 func (dc *DeploymentController) cleanupOldReplicaSets(oldRSs []*extensions.ReplicaSet, deployment *extensions.Deployment) error {
+	if deployment.Spec.RevisionHistoryLimit == nil {
```

Comment:
> Would you add a comment explaining what this function does. Specifically what "CleanUp" entails and what qualifies a ReplicaSet as "Old"

Will do

> 

A bit scary that no RevisionHistoryLimit == unlimited, but that is what the spec says.

Not really. This has always been the behavior for Deployments prior to adding RevisionHistoryLimit. Which is reasonable because we don't want to mess with a user's deployment history. But eventually most common use cases involve keeping around just a few of the latest replica sets and not all so we will end up defaulting this to 2 or 3 at some point (i think there is an issue about it and it is also in the deployment roadmap).